### ISSUE TITLE
Citrinet_1024 config file 

### ISSUE THREAD

**Issue by snaaz21** [NVIDIA/NeMo#2441]
Hi,

I am trying to finetune the Citrinet_1024 nemo model first on 50K chunks of the Indian accent English-language dataset using 1 gpu.
Since [here](https://github.com/NVIDIA/NeMo/tree/main/examples/asr/conf/citrinet) is no config file for citrinet_1024. So I changed the filters 512 to 1024, set the dropout to 0.01 and changed the model name from "Citrinet-512-8x-Stride" to "Citrinet-1024-8x-Stride".

Is that sufficient which I did above or need to change something else in the config file?

The config file which I modified is attached below in txt format;
[citrinet_1024.txt](https://github.com/NVIDIA/NeMo/files/6759687/citrinet_1024.txt)

**Comment by titu1994** [comment]
The easiest way is to load a Citrinet 1024 models, then save it's internal config with OmegaConf.save(model.cfg). You can then compare the difference. 

There's a few changes left, like enc_final filters being 1024 and changing batch size

**Comment by snaaz21** [comment]
I changed the enc_final filters to 1024 and batch_size to 32, and also tested by comparing it with the loaded config file of Citrinet_1024. The final config file is: [citrinet_1024.txt](https://github.com/NVIDIA/NeMo/files/6762899/citrinet_1024.txt)

**Comment by snaaz21** [comment]
While creating the tokenizers from asr dataset for citrinet_1024 finetuning, How to choose vocab size using the [script](https://github.com/NVIDIA/NeMo/blob/main/scripts/tokenizers/process_asr_text_tokenizer.py)?

`!python ./scripts/process_asr_text_tokenizer.py \
  --manifest="path/to/manifest.json" \
  --data_root="path/to/data_dir" \
  --vocab_size=?? \
  --tokenizer="wpe" \
  --no_lower_case \
  --log`

**Comment by titu1994** [comment]
1024 is sufficient for most languages

**Comment by snaaz21** [comment]
Hi, 

I fine-tuned the citrinet_1024 on 50k audio chunks for 2 epochs. But when I did inference using finetuned checkpoint (.nemo), it is not giving any output. 

The training script used for finetuning is --> [train.txt](https://github.com/NVIDIA/NeMo/files/6765918/train.txt) which I prepared with the help of [08_ASR_with_Subword_Tokenization](https://github.com/NVIDIA/NeMo/blob/main/tutorials/asr/08_ASR_with_Subword_Tokenization.ipynb) and [10_ASR_CTC_Language_Finetuning](https://github.com/NVIDIA/NeMo/blob/main/tutorials/asr/10_ASR_CTC_Language_Finetuning.ipynb). 

Please check once the training script which I prepared is correct or not? Or do I need to wait for more epochs?

**Comment by titu1994** [comment]
You will probably need to change the learning rate to be lower, plus since your vocabulary size is the same as the model's, you can perform the decoder weight loading step shown in the notebook to rapidly speed up convergence

**Comment by titu1994** [comment]
Also 50,000 chunks as in how many hours of speech is that? It might be beneficial for short runs to freeze the encoder - unfreeze batch norm ad shown to prevent the model needing to be trained in it's entirety

**Comment by snaaz21** [comment]
> Also 50,000 chunks as in how many hours of speech is that? It might be beneficial for short runs to freeze the encoder - unfreeze batch norm ad shown to prevent the model needing to be trained in it's entirety

not sure in hours, chunks are in between 2 to 15 sec. But in future I am planning to finetune the model on 6k-15k hours of English asr dataset (2-17 sec chunks).

**Comment by titu1994** [comment]
That's plenty of data no need to freeze encoder then (though you could try it if you wanted to). 

I think the decoder weight loading will generally help you significantly. The short run + high learning rate of the default in the model might have been the cause of this

**Comment by snaaz21** [comment]
> That's plenty of data no need to freeze encoder then (though you could try it if you wanted to).
> 
> I think the decoder weight loading will generally help you significantly. The short run + high learning rate of the default in the model might have been the cause of this

ok @titu1994, I'll be checking it.
Thank you so much for your quick replies.
